proxyd/backend.go

```diff
@@ -908,7 +911,31 @@ func (bg *BackendGroup) ProcessMulticallResponses(ch chan *multicallTuple, ctx c
 				"response_count", i,
 			)
 			if i > 0 {
-				return finalResp
+				// either all responses are failed, or all contain rpc errors
+
+				// check if no success response received
+				if len(successResp) == 0 {
+					return errResp
+				}
+
+				// add rpc error responses to success responses
+				for id, rpcRes := range rpcResToCheck {
+					if _, ok := successResp[id]; !ok {
+						successResp[id] = rpcRes
+					}
+				}
```

Comment:
Whats the goal here? Is it to add errored responses to the successful responses map that match the same ID?

If all the requests have a similar ID of 1, it may not work as intended

Reply:
Oh I see `isValidMulticallTx()` now.
I think that should work then if it's just 1 request each time, it would just only go through the loop once per check.